shared_model/builders/protobuf/builder_templates/query_template.hpp

```diff
@@ -0,0 +1,222 @@
+          query->set_account_id(account_id);
+        });
+      }
+
+      auto getSignatories(
+          const interface::types::AccountIdType &account_id) const {
+        return queryField([&](auto proto_query) {
+          auto query = proto_query->mutable_get_account_signatories();
+          query->set_account_id(account_id);
+        });
+      }
+
+      auto getAccountTransactions(
+          const interface::types::AccountIdType &account_id) const {
+        return queryField([&](auto proto_query) {
+          auto query = proto_query->mutable_get_account_transactions();
+          query->set_account_id(account_id);
+        });
+      }
+
+      auto getAccountAssetTransactions(
+          const interface::types::AccountIdType &account_id,
+          const interface::types::AssetIdType &asset_id) const {
+        return queryField([&](auto proto_query) {
+          auto query = proto_query->mutable_get_account_asset_transactions();
+          query->set_account_id(account_id);
+          query->set_asset_id(asset_id);
+        });
+      }
+
+      auto getAccountAssets(
+          const interface::types::AccountIdType &account_id,
+          const interface::types::AssetIdType &asset_id) const {
+        return queryField([&](auto proto_query) {
+          auto query = proto_query->mutable_get_account_assets();
+          query->set_account_id(account_id);
+          query->set_asset_id(asset_id);
+        });
+      }
+
+      auto getAccountDetail(const interface::types::AccountIdType &account_id,
+                            const interface::types::DetailType &detail) {
+        return queryField([&](auto proto_query) {
+          auto query = proto_query->mutable_get_account_detail();
+          query->set_account_id(account_id);
+          query->set_detail(detail);
+        });
+      }
+
+      auto getRoles() const {
+        return queryField(
+            [&](auto proto_query) { proto_query->mutable_get_roles(); });
+      }
+
+      auto getAssetInfo(const interface::types::AssetIdType &asset_id) const {
+        return queryField([&](auto proto_query) {
+          auto query = proto_query->mutable_get_asset_info();
+          query->set_asset_id(asset_id);
+        });
+      }
+
+      auto getRolePermissions(
+          const interface::types::RoleIdType &role_id) const {
+        return queryField([&](auto proto_query) {
+          auto query = proto_query->mutable_get_role_permissions();
+          query->set_role_id(role_id);
+        });
+      }
+
+      auto getTransactions(
+          std::initializer_list<interface::Transaction::HashType> hashes)
+          const {
+        return getTransactions(hashes);
+      }
+
+      template <typename... Hash>
+      auto getTransactions(const Hash &... hashes) const {
+        return getTransactions({hashes...});
+      }
+
+      template <typename Collection>
+      auto getTransactions(const Collection &hashes) const {
```

Comment:
This method should be moved above other two `getTransactions`, because they use this method.